apps/worker/src/app/workflow/services/trigger-worker.service.ts

```diff
@@ -1,16 +1,19 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, Logger } from '@nestjs/common';
 const nr = require('newrelic');
 import {
-  TriggerQueueService,
+  INovuWorker,
   PinoLogger,
   storage,
   Store,
   TriggerEvent,
   TriggerEventCommand,
+  TriggerQueueService,
 } from '@novu/application-generic';
 
+const LOG_CONTEXT = 'TriggerWorkerService';
+
 @Injectable()
-export class TriggerProcessorQueueService extends TriggerQueueService {
+export class TriggerWorkerService extends TriggerQueueService implements INovuWorker {
```

Comment:
Renaming as here we are instantiating a worker from BullMQ so for me the naming was confusing. Opinionated and debatable. Can roll back if needed.

Reply:
The old name was like that because at some point we had two queues for the trigger: producer and processor. 
I feel like the name should be similar to what we have for the other queues `WorkflowQueueService, WsQueueService`, so like `TriggerQueueService`, but I'm not sure why we have `TriggerQueueService` and not are reusing the `QueueService`

Reply:
I can give it a look regarding not using QueueService. But I still think using Queue when we are instantiating a worker is confusing. Maybe here I am thinking more in AWS terms of architecture (SQS, SNS). Also we can raise that as an open question in the team. What do you think?

Reply:
You worded nicely @djabarovgeorge. Indeed it is the terminology that I was aiming with these changes as latest years have worked a lot with Pub/Sub tools and patterns and I personally enjoy that separation of concerns. I also see we have that separation in our implementation (it can be seen in the services created and point of discussion) so to me made sense heading that direction with the naming. But again, something worth to bring to the team.

Reply:
I have been pairing with @LetItRock and we agreed to tackle the proposed refactor in a different PR, as the separation of concerns is a good thing and we can have it less confusing as right now.